### DISCUSSION TITLE
Private gitlab server: forge works for a few minutes then stops working (help debugging)

### DISCUSSION THREAD

**Discussion by angel-devicente** [magit/forge#582] [Support] [answered]
Hello,
I set Magit+Forge to work with a local Gitlab server. Things seem to be properly configured, since I can get the list of all issues in the repository and then create a new one without problems. But after only a few minutes, creating a new issue doesn't work (I don't get any error either, just a message that [....]/posts/new-issue was written, but the buffer does not get closed and if I go to the Magit status buffer, that issue certainly was not added.

What could be the problem? It looks like a timeout issue, but not sure where to look and/or how to debug it. 

Any pointers? Thanks

**Comment by tarsius** [comment]
```elisp
(defun forge-post-submit ()
  "Submit the post that is being edited in the current buffer."
  (interactive)
  (save-buffer)
  (if-let ((fn forge--submit-post-function))
      (funcall fn
               (forge-get-repository forge--buffer-post-object)
               forge--buffer-post-object)
    (error "forge--submit-post-function is nil")))
```

> just a message that [....]/posts/new-issue was written

The `save-buffer` above does that, while `forge--submit-post-function` is this:

```elisp
(cl-defmethod forge--submit-create-issue ((_ forge-gitlab-repository) repo)
  (let-alist (forge--topic-parse-buffer)
    (forge--glab-post repo "/projects/:project/issues"
      `((title       . , .title)
        (description . , .body))
      :callback  (forge--post-submit-callback)
      :errorback (forge--post-submit-errorback))))
```

Jumping along the call stack we get to `ghub-request`. Also note the callbacks above. Add debug statements to them. Are they ever called?

I suspect that what is happening is that your companies gitlab for whatever reason does not respond to the request but also does not close the connection. I think `url.el` only supports timeouts for synchronous requests, but this is asynchronous, so we wait forever (or just longer than you have so far).

I don't think there is much I can to to diagnose this from a distance.

**Reply by angel-devicente** [reply, marked as answer]
Thanks for the help. 

I will try at some point to diagnose the problem, but the truth is that our company server was giving me other troubles as well, so for the time being I've migrated the repositories to Github, and magit/forge works really nice there.
